### PR TITLE
Feature/zcnsc eventsdb storage

### PR DESCRIPTION
**Pull request by Dmdv**
This PR adds events emitter to `AddAuthorizer` and `DeleteAuthorizer` functions plus mocks updates and events helper

Tasks to complete before merging PR:
- [ ]  Ensure system tests are passing. If not [Run them manually](https://github.com/0chain/0chain/actions/workflows/system_tests.yml) to check for any regressions :clipboard:
- [ ]  Do any new system tests need added to test this change? do any existing system tests need updated? If so create a PR at [0chain/system_test](https://github.com/0chain/system_test)
- [ ]  Merge your system tests PR to master BEFORE merging this PR  
  
Associated PRs (Link as appropriate):
...

### CODE REVIEW

**Comment by Sriep** [comment]
We should have a getAuthoriserNodes REST API endpoint to use the eventDb.

**Comment by Dmdv** [comment]
> We should have a getAuthoriserNodes REST API endpoint to use the eventDb.

@Sriep Fixed it

**Comment by Sriep** [comment]
It is looking much better. Some points.
1. Do we need to worry about backward compatibility? I mean are people using getAuthorizerNodes anywhere? Probably not, but if so we would need to maintain the pre-existing method of getting Authorisers from the MPT.
2. How are people going to use /getAuthorizerNodes and /getAuthorizerNode? 
3. /getAuthorizerNodes will likely return too much data if we return all authorises.
4. Don't we want changes to GoSDK and zwallet(or zbox) to access the  `/getAuthorizerNode` and ` /getAuthorizerNodes` endpoints. This will allow testing.
5. Should we be writing system tests for the getAuthorzier endpoints?

**Comment by service-0chain** [comment]
Manual system tests [failure] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | feature/default_client_config  |
| gosdk  | feature/default_client_config  |
| zboxcli  | staging  |
| zwalletcli  | feature/update_74  |
| 0chain  | feature/zcnsc_eventsdb_storage  |
| blobber  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Comment by Dmdv** [comment]
> It is looking much better. Some points.
> 
> 1. Do we need to worry about backward compatibility? I mean are people using getAuthorizerNodes anywhere? Probably not, but if so we would need to maintain the pre-existing method of getting Authorisers from the MPT.
> 2. How are people going to use /getAuthorizerNodes and /getAuthorizerNode?
> 3. /getAuthorizerNodes will likely return too much data if we return all authorises.
> 4. Don't we want changes to GoSDK and zwallet(or zbox) to access the  `/getAuthorizerNode` and ` /getAuthorizerNodes` endpoints. This will allow testing.
> 5. Should we be writing system tests for the getAuthorzier endpoints?

1. We don't need backward compatibility in this case
2. `/getAuthorizerNodes` won't be used by any type of client, it's used in utils, zwalletcli
3. It will be returning no more than 100 authorizers {ID, URL}
4. It is implemented in GOSDK
5. System tests are written to list authorizers

@Sriep

File: code/go/0chain.net/smartcontract/zcnsc/benchmark_rest_tests.go [hunk 1]
```diff
@@ -30,21 +30,21 @@ func (bt restBenchTest) Run(balances cstate.StateContextI, _ *testing.B) error {
...
-				name:     "zcnsc_rest.getAuthorizerNode",
-				endpoint: sc.getAuthorizerNode,
-				params: func() url.Values {
-					var values url.Values = make(map[string][]string)
-					values.Set("id", auth.ID)
-					return values
-				}(),
+				name:     "zcnsc_rest.getAuthorizerNodes",
+				endpoint: sc.getAuthorizerNodes,
+				//params: func() url.Values {
```
Comment (Sriep) [inline code/go/0chain.net/smartcontract/zcnsc/benchmark_rest_tests.go]: remove commented out code.

File: code/go/0chain.net/smartcontract/zcnsc/handler.go [hunk 2]
```diff
@@ -2,24 +2,52 @@ package zcnsc
...
-func (zcn *ZCNSmartContract) getAuthorizerNode(_ context.Context, params url.Values, ctx cState.StateContextI) (interface{}, error) {
-	authorizerID := params.Get("id")
-	if authorizerID == "" {
-		return nil, errors.New("authorizerID is empty")
+type AuthorizerNodeResponse struct {
+	ID  string `json:"id"`
+	URL string `json:"url"`
+}
+
+func (zcn *ZCNSmartContract) getAuthorizerNodes(
```
Comment (Sriep) [inline code/go/0chain.net/smartcontract/zcnsc/handler.go]: In anticipation of https://github.com/0chain/0chain/issues/926 we could prefix new endpoints with  goDoc quality comments.

**Review by Sriep — DISMISSED** [review]

File: code/go/0chain.net/smartcontract/zcnsc/benchmark_rest_tests.go (hunk 1 above)
Comment (Dmdv) [reply code/go/0chain.net/smartcontract/zcnsc/benchmark_rest_tests.go]: fixed

File: code/go/0chain.net/smartcontract/zcnsc/handler.go (hunk 2 above)
Comment (Dmdv) [reply code/go/0chain.net/smartcontract/zcnsc/handler.go]: fixed

**Comment by service-0chain** [comment]
Manual system tests [success] with the following config
| config  | value |
| ------------- | ------------- |
| tests skipped  | FALSE  |
| system-tests  | master  |
| gosdk  | NOT OVERRIDDEN  |
| zboxcli  | staging  |
| zwalletcli  | staging  |
| 0chain  | staging  |
| blobber  | staging  |
| authorizer  | staging  |
| 0box  | staging  |
| 0block  | staging  |
| 0dns  | staging  |
| explorer  | staging  |
| 0proxy  | staging  |
| 0search  | staging  |

**Pull request merged**